#!/usr/bin/env python3
PACKAGE = "placement_dynamic_reconfigure"

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()
gen.add("frame", str_t, 0, "frame", "map")
gen.add("target_point_x", double_t, 0, "target point x", 0.0, -5.0, 5.0)
gen.add("target_point_y", double_t, 0, "target point y", 0.0, -5.0, 5.0)
gen.add("target_point_z", double_t, 0, "target point z", 0.55, -5.0, 5.0)

gen.add("box_filter_range_x", double_t, 0, "box filter range x", 4.0, 0.0, 5.0)
gen.add("box_filter_range_y", double_t, 0, "box filter range y", 4.0, 0.0, 5.0)
gen.add("box_filter_range_z", double_t, 0, "box filter range z", 1.0, 0.0, 5.0)

gen.add("vertical_axis_x", double_t, 0, "vertical axis x", 0.0, 0.0, 5.0)
gen.add("vertical_axis_y", double_t, 0, "vertical axis y", 0.0, 0.0, 5.0)
gen.add("vertical_axis_z", double_t, 0, "vertical axis z", 1.0, 0.0, 5.0)

gen.add("tilt_threshold", double_t, 0, "tilt threshold in degrees", 60.0, 0.0, 360.0)

gen.add("distance_threshold", double_t, 0, "distance threshold", 1.0, 0.0, 5.0)

gen.add("obj_dim_x", double_t, 0, "object dimension x", 0.10, 0.0, 1.0)
gen.add("obj_dim_y", double_t, 0, "object dimension y", 0.10, 0.0, 1.0)
gen.add("obj_dim_z", double_t, 0, "object dimension z", 0.10, 0.0, 1.0)

gen.add("obj_to_surface_x", double_t, 0, "object to surface x", 0.10, 0.0, 1.0)
gen.add("obj_to_surface_y", double_t, 0, "object to surface y", 0.10, 0.0, 1.0)
gen.add("obj_to_surface_z", double_t, 0, "object to surface z", 0.10, 0.0, 1.0)
gen.add("obj_to_surface_quat_x", double_t, 0, "object to surface quaternion x", 0.0, 0.0, 1.0)
gen.add("obj_to_surface_quat_y", double_t, 0, "object to surface quaternion y", 0.0, 0.0, 1.0)
gen.add("obj_to_surface_quat_z", double_t, 0, "object to surface quaternion z", 0.0, 0.0, 1.0)
gen.add("obj_to_surface_quat_w", double_t, 0, "object to surface quaternion w", 1.0, 0.0, 1.0)

gen.add("surface_range", double_t, 0, "surface range", 1.0, 0.0, 5.0)

exit(gen.generate(PACKAGE, "test_placement", "placement"))
